### PR TITLE
Allow tagging of source user on Quip.

### PR DESCRIPTION
**Pull request by hashamali**
## Overview

Quips are now some thangs. User can optionally assign a source to a Quip as well as which circle to share it to.
## Technical
- Move InviteFriendsFragment to use SearchListFragment.
- Create activity for Quip creation that has a FriendSearchListFragment for selecting the source, and another activity for circle selection using FriendsSearchListFragment.
- Add activity transitions.
## Reviewers
- [ ] @dbuscaglia 
- [ ] @jcomo

### CODE REVIEW

File: app/src/main/java/it/quip/android/activity/CreateQuipActivity.java [hunk 1]
```diff
@@ -80,26 +79,47 @@ private void onPostQuip() {
...
+        }
+
+        long timestamp = TimeUtils.currentTimestampInS();
 
-        showShareFragment();
+        if (circles.size() > 0) {
+            for (Circle circle : circles) {
+                Quip newQuip = new Quip(quip);
+                newQuip.setCircle(circle);
+                newQuip.setTimestamp(timestamp);
```
Comment (jcomo) [inline app/src/main/java/it/quip/android/activity/CreateQuipActivity.java]: Remove the call to setTimestamp here, we took that out

File: app/src/main/java/it/quip/android/activity/CreateQuipActivity.java [hunk 2]
```diff
@@ -80,26 +79,47 @@ private void onPostQuip() {
...
+                newQuip.saveInternal();
+            }
+        } else {
+            quip.setTimestamp(timestamp);
```
Comment (jcomo) [inline app/src/main/java/it/quip/android/activity/CreateQuipActivity.java]: And here

File: app/src/main/java/it/quip/android/activity/ShareQuipActivity.java [hunk 3]
```diff
@@ -1,39 +1,66 @@
...
+        super.onCreate(savedInstanceState);
+
+        setupDependencies();
+        setupView();
+        showCircleSelectFragment();
+    }
+
+    public static List<Circle> getCircles() {
+        List<Circle> result = sCircles;
+        sCircles = null;
```
Comment (jcomo) [inline app/src/main/java/it/quip/android/activity/ShareQuipActivity.java:61]: Why set this to null immediately?

**Comment by jcomo** [comment]
#thangs :+1:

File: app/src/main/java/it/quip/android/activity/ShareQuipActivity.java (hunk 3 above)
Comment (hashamali) [inline app/src/main/java/it/quip/android/activity/ShareQuipActivity.java:61]: Just to ensure there is not errant dangling pointer to old data. Its a static reference so I want to be sure it doesn't pick up old data the next time its used.

**Pull request merged**